src/credentials/FabricTable.h

```diff
@@ -993,6 +1037,10 @@ class DLL_EXPORT FabricTable
     CHIP_ERROR UpdatePendingFabricCommon(FabricIndex fabricIndex, const ByteSpan & noc, const ByteSpan & icac,
                                          Crypto::P256Keypair * existingOpKey, bool isExistingOpKeyExternallyOwned);
 
+    // Common code for looking up a fabric given a root public key, a fabric ID and an optional node id scoped to that fabric.
+    const FabricInfo * FindFabricCommon(const Crypto::P256PublicKey & rootPubKey, FabricId fabricId,
+                                        NodeId nodeId = kUndefinedNodeId) const;
```

Comment:
You need to add missing unit test coverage for the new usage, especially making sure that adding colliding entries leads to being able to find the right identitiies with `FindIdentity`

Reply:
Added a new `TestCollidingFabrics` test to `TestFabricTable` that validates loading multiple NOCs on the same fabric into the fabric table in both cases of allowing/disallowing collisions in the table. I also then validated FindIdentity.